docs/guide/uploading-downloading-binary-content.md

```diff
@@ -0,0 +1,147 @@
+| [`$delete()`](/docs/reference.md#delete) | Permanently deletes the file stored at the configured path. Returns a promise that is resolved when the delete completes. |
+| [`$toString()`](/docs/reference.md#tostring) | Returns a string version of the bucket path stored as a `gs://` scheme. |
+
+
+## Uploading files
+To upload files use either the `$put()` or `$putString()` methods. These methods
+return an `UploadTask` which is wrapped by AngularFire to handle the `$digest` loop.
+
+```js
+function SampleCtrl($firebaseStorage) {
+  // create a Firebase Storage Reference for the $firebaseStorage binding
+  var storageRef = firebase.storage().ref('userProfiles/physicsmarie');
+  var storage = $firebaseStorage(storageRef);
+  var file = // get a file from the template (see Retrieving files from template section below)
+  var uploadTask = storage.$put(file);
+  // of upload via a RAW, base64, or base64url string
+  var stringUploadTask = storage.$putString('5b6p5Y+344GX44G+44GX44Gf77yB44GK44KB44Gn44Go44GG77yB', 'base64');
+}
+SampleCtrl.$inject = ["$firebaseStorage"];
+```
+
+### Upload Task API Summary
+
+| Method  | Description |
+| ------------- | ------------- |
+| [`$progress(callback)`](/docs/reference.md#uploadtask-progress) |	Calls the provided callback function whenever there is an update in the progress of the file uploading. |
+| [`$error(callback)`](/docs/reference.md#uploadtask-error)	| Calls the provided callback function when there is an error uploading the file |
+| [`$complete(callback)`](/docs/reference.md#uploadtask-complete) |	Calls the provided callback function when the upload is complete. |
+| [`$cancel()`](/docs/reference.md#uploadtask-cancel) | Cancels the upload. |
+| [`$pause()`](/docs/reference.md#uploadtask-pause) | Pauses the upload. |
+| [`$snapshot()`](/docs/reference.md#uploadtask-$snapshot) | Returns the [current immutable view of the task](https://firebase.google.com/docs/storage/web/upload-files#monitor_upload_progress) at the time the event occurred. |
+| [`then(callback)`](/docs/reference.md#uploadtask-then) | An Upload Task implements a promise like interface. The callback is called when the upload is complete. |
+| [`catch(callback)`](/docs/reference.md#uploadtask-then) | An Upload Task implements a promise like interface. The callback is called when an error occurs. |
+
+## Displaying images with the `firebase-src` directive
+AngularFire provides a directive that displays a file with any `src` compatible element.
+
+Instead of using the tradional `src` attribute, use `firebase-src`:
+
+```html
+<img firebase-src="userProfiles/physicsmarie" />
+<!-- Works with bindings as well -->
+<img firebase-src="{{ userProfileId }}" />
+```
+
+## Retrieving files from the template
+AngularFire does not provide a directive for retrieving a upload file. However,
+the below directive is basic but provides a baseline to work off.
```

Comment:
Suggestion: get rid of "is basic but" and replace the closing period with a colon.